crates/primitives/src/transaction/mod.rs

```diff
@@ -173,6 +194,10 @@ impl Transaction {
         match self {
             Transaction::Legacy(_) | Transaction::Eip2930(_) => false,
             Transaction::Eip1559(_) | Transaction::Eip4844(_) => true,
+            Transaction::Eip1559(TxEip1559 { gas_limit, .. }) => *gas_limit,
+            Transaction::Eip1559(TxEip1559 { gas_limit, .. }) => *gas_limit,
+            #[cfg(feature = "optimism")]
+            Transaction::Deposit(TxDeposit { gas_limit, .. }) => *gas_limit,
```

Comment:
Should return the boolean value.
```suggestion
            #[cfg(feature = "optimism")]
            Transaction::Deposit(_) => true,
```